### ISSUE TITLE
[BUG] Grouping is not automatically used in qnode execution (or tracker is wrong)

### ISSUE THREAD

**Issue by Qottmann** [PennyLaneAI/pennylane#5561]
**Expected behavior**

```python3
shots = 20

H_qwc = qml.sum(0.5 * X(0), 0.5 * X(0) @ X(1), grouping_type="qwc")
print("groupings: ", H_qwc.grouping_indices)

dev_finite = qml.device("default.qubit", shots=shots)

# @qml.transforms.hamiltonian_expand
@qml.qnode(dev_finite)
def qnode_finite(H_qwc):
    return qml.expval(H_qwc)

with qml.Tracker(dev_finite) as tracker_finite:
    res_finite = qnode_finite(H_qwc)

tracker_finite.totals["shots"]
```

```pycon
groupings: ((0, 1),)
20 # shots * 1
```

**Actual behavior** 

```pycon
groupings: ((0, 1),)
40 # shots * 2
```

Here, both observables are qubit-wise-commuting, so they should be able to me measured simultaneously with one execution.
It seems the grouping is not used in the qnode execution, and it simply computes all terms separately.

**Context**

So far grouping was reduced to `Hamiltonian` instances but since `v0.35` also `Sum` objects can have it. In that sense, it is a relatively new feature. However, with new opmath enabled, molecular hamiltonians are now actually `Sum` instances, so this is more relevant than it might seem at first.

**Comment by albi3ro** [comment]
Wasn't this fixed by #5525  @astralcai ?

**Comment by astralcai** [comment]
> Wasn't this fixed by #5525 @astralcai ?

I thought so, let me look into it.

**Comment by astralcai** [comment]
Seems that the tracking is incorrect, let me fix that.
